src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -330,6 +344,12 @@ instance Buildable Direction where
 newtype SignedTx = SignedTx { signedTx :: ByteString }
     deriving (Show, Eq, Generic)
 
+data TxWitness
+    = PublicKeyWitness ByteString (Hash "tx")
+    | ScriptWitness ByteString
+    | RedeemWitness ByteString
```

Comment:
We only _care_ about PublicKey witness, others are irrelevant to us. For now at least.